src/Emulator/Main/Peripherals/Timers/ComparingTimer.cs

```diff
@@ -143,6 +143,11 @@ public virtual void Reset()
             InternalReset();
         }
 
+        public virtual void ClearCounterValue()
```

Comment:
I'm not sure if this is (a) necessary and (b) correct (assuming that your intention is to clear the value of the timer).

Timer's value is calculated by summing `valueAccumulatedSoFar` with the current value of the `clockEntry` (this is an object we use to keep track of time internally in Renode) because we need to keep track of two limits - the maximal value at which the counter wraps to 0 and the compare value that triggers an event. You can take a look at the code and comments in `ComparingTimer.ComapreReachedInternal()` if you feel adventurous ;)

Anyways, I believe that setting `Value` property to 0 should work.

Reply:
yeah, I didn't understand that - thanks for your help here!